lib/msal-common/src/app/module/AuthorizationCodeModule.ts

```diff
@@ -111,6 +111,36 @@ export class AuthorizationCodeModule extends AuthModule {
     }
 
     async acquireToken(request: TokenExchangeParameters): Promise<TokenResponse> {
+        const acquireTokenAuthority = (request && request.authority) ? AuthorityFactory.createInstance(request.authority, this.networkClient) : this.defaultAuthorityInstance;
+
+        let tokenEndpoint: string;
+        try {
+            tokenEndpoint = acquireTokenAuthority.tokenEndpoint;
```

Comment:
if there is a default authority why does this throw?

Reply:
When we come back from a redirect, the default authority does not resolveEndpoints automatically, so we need to check if we still have the endpoint available or we need to resolve again.

Reply:
So `createInstance` can return `null`?

Reply:
Would be simpler in terms of syntax to check the truthiness of `acquireTokenAuthority`, then to do a `try/catch`.

Reply:
`createInstance` won't return null. Essentially authority creation has two steps:

1. Create the authority object with no data for oauth endpoints (i.e. /authorize, /token, etc,).
2. Resolve the /.well-known oauth endpoint to retrieve required data for authority.

Checking the truthiness of acquireTokenAuthority doesn't tell us whether we have done (2). there is a private property in the authority class which DOES denote this, so I can change this code to check that parameter, but it essentially accomplishes the same thing.

Reply:
Oh, I see the getter below for `tokenEndpoint`.

Reply:
resolving until we make changes for endpoint discovery